### ISSUE TITLE
visual_behavior.data_access.loading.get_filtered_ophys_session_table is failing

### ISSUE THREAD

**Issue by dougollerenshaw** [AllenInstitute/visual_behavior_analysis#669]
```
import visual_behavior.data_access.loading as loading
session_table = loading.get_filtered_ophys_session_table()
```
returns:
```
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-6-86af1241af0e> in <module>
      1 import visual_behavior.data_access.loading as loading
      2 
----> 3 session_table = loading.get_filtered_ophys_session_table()

~/Code/visual_behavior_analysis/visual_behavior/data_access/loading.py in get_filtered_ophys_session_table()
    212     sessions = filtering.limit_to_production_project_codes(sessions)
    213     sessions = filtering.limit_to_valid_ophys_session_types(sessions)
--> 214     sessions = reformat.add_all_qc_states_to_ophys_session_table(sessions)
    215     sessions = filtering.limit_to_passed_ophys_sessions(sessions)
    216     sessions = filtering.remove_failed_containers(sessions)

~/Code/visual_behavior_analysis/visual_behavior/data_access/reformat.py in add_all_qc_states_to_ophys_session_table(session_table)
     88             :return: session_table: with additional columns added
     89             """
---> 90     experiment_table = loading.get_filtered_ophys_experiment_table(include_failed_data=True)
     91     session_table = add_session_workflow_state_to_ophys_session_table(session_table, experiment_table)
     92     session_table = add_container_workflow_state_to_ophys_session_table(session_table, experiment_table)

~/Code/visual_behavior_analysis/visual_behavior/data_access/loading.py in get_filtered_ophys_experiment_table(include_failed_data)
    169         experiments = filtering.limit_to_valid_ophys_session_types(experiments)
    170         experiments = filtering.remove_failed_containers(experiments)
--> 171     experiments['session_number'] = [int(session_type[6]) for session_type in experiments.session_type.values]
    172     experiments = experiments.drop_duplicates(subset='ophys_experiment_id')
    173     experiments = experiments.set_index('ophys_experiment_id')

~/Code/visual_behavior_analysis/visual_behavior/data_access/loading.py in <listcomp>(.0)
    169         experiments = filtering.limit_to_valid_ophys_session_types(experiments)
    170         experiments = filtering.remove_failed_containers(experiments)
--> 171     experiments['session_number'] = [int(session_type[6]) for session_type in experiments.session_type.values]
    172     experiments = experiments.drop_duplicates(subset='ophys_experiment_id')
    173     experiments = experiments.set_index('ophys_experiment_id')

ValueError: invalid literal for int() with base 10: 'i'

```

**Comment by dougollerenshaw** [comment]
The reason for the failure is that a 'VisCodingTargetedMovieClips' experiment is in the table. 

![image](https://user-images.githubusercontent.com/19944442/92265782-3f223b80-ee94-11ea-9b48-85ba80194c0c.png)

The session number is expected to be at character 6, assuming the 'OPHYS_{}_xxx' format. 

How did this experiment type end up in the table? This is a new problem.

**Comment by dougollerenshaw** [comment]
Still narrowing this down. I've narrowed the issue down to this line:

https://github.com/AllenInstitute/visual_behavior_analysis/blob/a3ab84fbd5ef0ed56663b1b17c5b6ce92b0517cc/visual_behavior/data_access/loading.py#L171

Which was just added to `get_filtered_ophys_experiment_table` as part of PR #667

The line was added by @matchings in commit https://github.com/AllenInstitute/visual_behavior_analysis/commit/424572893dcc36dd636cf7cd279bebfade68c0dd

However, there are 8 sessions with the session_type = 'VisCodingTargetedMovieClips', which cause this line to crash:
![image](https://user-images.githubusercontent.com/19944442/92287741-82de6a80-eebf-11ea-8cbf-723fabd35ed1.png)

**Comment by dougollerenshaw** [comment]
All 8 'VisCodingTargetedMovieClips' experiments came from a single mesoscope session:
![image](https://user-images.githubusercontent.com/19944442/92288054-7a3a6400-eec0-11ea-8535-4af3ef21fe95.png)

**Comment by dougollerenshaw** [comment]
Just to be a bit more explicit, here's minimal code that returns the experiments above. 

```
import visual_behavior.data_access.loading as loading
cache = loading.get_visual_behavior_cache()
experiments = cache.get_experiment_table()
cols_to_display = [
    'project_code',
    'experiment_workflow_state',
    'equipment_name',
    'ophys_session_id',
    'session_name',
    'session_type',
    'date_of_acquisition',
]
experiments.query('session_type == "VisCodingTargetedMovieClips"')[cols_to_display]
```

![image](https://user-images.githubusercontent.com/19944442/92288414-b4f0cc00-eec1-11ea-94f8-0c1acf5269d2.png)

**Comment by matchings** [comment]
@dougollerenshaw ok i think i have this figured out.

First, i added an exception to the line that creates the `session_number `column by taking the 6th index so that it will now return None if the string 'images' is not in `session_type`. This will prevent it from giving an error if receptive field mapping sessions such as 'VisCodingTargetedMovieClips' are included in the `experiments_table`. 

Second, the call to filter out invalid session types (including receptive field mapping sessions), `sessions = filtering.limit_to_valid_ophys_session_types(sessions)` , was called before a function that added columns from the unfiltered experiments table, resulting in the output of get_filtered_ophys_session_table() including some unwanted sessions (receptive field mapping sessions). i moved the call to `limit_to_valid_ophys_session_types(sessions)` down a line so that the unwanted sessions are properly filtered out. 

These changes are reflected in this PR: https://github.com/AllenInstitute/visual_behavior_analysis/pull/671

**Comment by dougollerenshaw** [comment]
Fixed in issue #671
